R/gen.R

```diff
@@ -0,0 +1,73 @@
+#' Create a function to draw from the posterior predictive distribution for a
+#' latent model
+#'
+#' This function creates a function that draws from the posterior predictive
+#' distribution for a latent model using [primarycensored::rpcens()] to handle
+#' censoring and truncation. The returned function takes a prep argument from
+#' `brms` and returns posterior predictions. This is used internally by
+#' [brms::posterior_predict()] to generate predictions for latent models.
+#'
+#' @inheritParams epidist_family_model
+#'
+#' @return A function that takes a prep argument from brms and returns a matrix
+#' of posterior predictions, with one row per posterior draw and one column
+#' per observation. The prep object must have the following variables:
+#' * `vreal1`: relative observation time
+#' * `vreal2`: primary event window
+#' * `vreal3`: secondary event window
+#'
+#' @seealso [brms::posterior_predict()] for details on how this is used within
+#' `brms`, [primarycensored::rpcens()] for details on the censoring approach
+#' @autoglobal
+#' @family gen
+#' @export
+epidist_gen_posterior_predict <- function(family) {
```

Comment:
Is this function specific to the latent model? If so I'd consider including it in the name

Reply:
I can see that the `log_lik` one has the name in. I don't think these functions will work by default with e.g. the marginal model. Is the point that they'll be modified then? Some question about whether for the time being for consistency they should be named `latent` given that they don't work with the marginal model as is (and the marginal model doesn't exist yet)?

Reply:
as discussed the idea is that some are generic to the data generating process, some are just generic (i.e the epred approach) and some are model specific (i.e this latent one). We could just stick them all in the model for but I would prefer to aim to generalise from the get go.

Reply:
I'll make an issue to explore how generalisable these are

Reply:
See #462